src/experimental/Toast/ToastUI.tsx

```diff
@@ -0,0 +1,145 @@
+import {CheckIcon, AlertIcon, StopIcon, InfoIcon, XIcon} from '@primer/octicons-react'
+
+import {Spinner, Box, IconButton} from '../..'
+
+const variants = ['default', 'success', 'attention', 'danger', 'loading'] as const
+
+const icon: Record<ToastVariants, React.ReactNode> = {
+  default: <InfoIcon size="small" />,
+  success: <CheckIcon size="small" />,
+  attention: <AlertIcon size="small" />,
+  danger: <StopIcon size="small" />,
+  loading: <Spinner size="small" />,
+}
+
+export const ToastUI = ({
+  dismissable = false,
+  variant = 'default',
+  role = 'status',
+  dismissLabel = 'Dismiss this message',
+  children,
+}: React.PropsWithChildren<ToastUIProps>) => {
+  if (__DEV__) {
+    // eslint-disable-next-line no-console
+    console.assert(
+      variants.includes(variant),
+      `The variant ${variant} is not supported. Use one of ${variants.join(', ')}`,
+    )
+  }
+
+  return (
+    <Box role={role} sx={{display: 'flex', justifyContent: 'flex-start'}}>
+      <Box
+        sx={{display: 'inline-flex', maxWidth: '450px', borderRadius: 2, boxShadow: 'shadow.large', overflow: 'hidden'}}
+      >
+        <Box
+          sx={{
+            display: 'flex',
+            justifyItems: 'center',
+            alignItems: 'center',
+            borderTopLeftRadius: 2,
+            borderBottomLeftRadius: 2,
+            border: '1px solid',
+            borderColor: 'border.default',
+            borderInlineEnd: 'none',
+            p: 3,
+            color: 'white',
+            '&[data-variant="default"]': {
+              backgroundColor: 'accent.fg',
+              borderColor: 'accent.muted',
+            },
+            '&[data-variant="success"]': {
+              backgroundColor: 'success.fg',
+              borderColor: 'success.muted',
+            },
+            '&[data-variant="attention"]': {
+              backgroundColor: 'attention.fg',
+              borderColor: 'attention.muted',
+            },
+            '&[data-variant="danger"]': {
+              backgroundColor: 'danger.fg',
+              borderColor: 'danger.muted',
+            },
+            '&[data-variant="loading"]': {
+              backgroundColor: 'neutral.emphasis',
+              borderColor: 'neutral.muted',
+            },
+          }}
+          role="presentation"
+          data-variant={variant}
+        >
+          {icon[variant]}
+        </Box>
+        <Box
+          sx={{
+            display: 'flex',
+            alignItems: 'center',
+            gap: 3,
+            justifyContent: 'space-between',
+            borderBottomRightRadius: 2,
+            borderTopRightRadius: 2,
+            border: '1px solid',
+            borderColor: 'border.default',
+            borderInlineStart: 'none',
+            px: 3,
+          }}
+        >
+          <Box
+            sx={{
+              /* we margin to allow for margin collapsing when user-land uses as the children <Text as="p" /> */
+              marginBlock: 3,
```

Comment:
A few things here:

- The `3` isn't mapping to a space variable (16px) I think because styled-components doesn't understand `marginBlock`. So its currently rendering 3px.
- I understand why you added this and it makes sense to have 16px of margin for the content, _but_ it looks like our Text component doesn't have any kind of browser reset. So by default it already has 1em of padding 😬 soooo.. maybe you could reset that in the scope of this component?

Reply:
great catch! I think @siddharthkp mentioned I could try my hand at css-modules, do you know how to get theme variables in those files?

Reply:
🤩 If you wanted to move towards a CSS file, you would access CSS variables instead which are a bit different! That can all be found in the [Primer Primitives Storybook](https://primer.style/primitives/storybook/?path=/story/size-base--base), but this is all very new so I don't have a ton of guidance on how to use specific properties. I'm happy to help though so if you give it a go, ping me again on this PR.

Reply:
Oh awesome! Wasn't aware that the new primitives was accessible in primer-react just yet. That's fantastic. Will give it a go!